apps/backend/src/app/api/admin/semesters/[id]/route.test.ts

```diff
@@ -38,16 +45,39 @@ describe("/api/admin/semesters/[id]", async () => {
       expect(await res.json()).toStrictEqual({ error: "No scope" })
     })
 
-    it("should delete semester if user is admin", async () => {
+    it("should delete semester and related documents if user is admin", async () => {
       cookieStore.set(AUTH_COOKIE_NAME, adminToken)
       const newSemester = await semesterDataService.createSemester(semesterCreateMock)
+      const newGameSessionSchedule = await gameSessionDataService.createGameSessionSchedule({
+        ...gameSessionScheduleMock,
+        semester: newSemester,
+      })
+      const newGameSession = await gameSessionDataService.createGameSession({
+        ...gameSessionMock,
+        gameSessionSchedule: newGameSessionSchedule,
+      })
+      const newBooking = await bookingDataService.createBooking({
+        ...bookingCreateMock,
+        gameSession: newGameSession,
+      })
 
       const res = await DELETE({} as NextRequest, {
         params: Promise.resolve({ id: newSemester.id }),
       })
 
       expect(res.status).toBe(StatusCodes.NO_CONTENT)
-      await expect(semesterDataService.getSemesterById(newSemester.id)).rejects.toThrow("Not Found")
+      await expect(semesterDataService.getSemesterById(newSemester.id)).rejects.toThrow(
+        getReasonPhrase(StatusCodes.NOT_FOUND),
+      )
+      await expect(
+        gameSessionDataService.getGameSessionScheduleById(newGameSessionSchedule.id),
+      ).rejects.toThrow(getReasonPhrase(StatusCodes.NOT_FOUND))
+      await expect(gameSessionDataService.getGameSessionById(newGameSession.id)).rejects.toThrow(
+        getReasonPhrase(StatusCodes.NOT_FOUND),
+      )
+      await expect(bookingDataService.getBookingById(newBooking.id)).rejects.toThrow(
+        getReasonPhrase(StatusCodes.NOT_FOUND),
+      )
```

Comment:
this is not how we assert data-layer `NotFound`

`getReasonPhrase(StatusCodes.NOT_FOUND)` is used for route based HTTP not founds

```suggestion
      await expect(semesterDataService.getSemesterById(newSemester.id)).rejects.toThrow("Not Found")
      await expect(
        gameSessionDataService.getGameSessionScheduleById(newGameSessionSchedule.id),
      ).rejects.toThrow("Not Found")
      await expect(gameSessionDataService.getGameSessionById(newGameSession.id)).rejects.toThrow("Not Found")
      await expect(bookingDataService.getBookingById(newBooking.id)).rejects.toThrow("Not Found")
```